src/NHibernate/Id/Insert/AbstractSelectingDelegate.cs

```diff
@@ -26,57 +26,67 @@ protected internal AbstractSelectingDelegate(IPostInsertIdentityPersister persis
 
 		public abstract IdentifierGeneratingInsert PrepareIdentifierGeneratingInsert();
 
-		public object PerformInsert(SqlCommandInfo insertSQL, ISessionImplementor session, IBinder binder)
+		public object PerformInsert(SqlCommandInfo insertSql, ISessionImplementor session, IBinder binder)
 		{
+			session.ConnectionManager.FlushBeginning();
 			try
 			{
-				// prepare and execute the insert
-				var insert = session.Batcher.PrepareCommand(insertSQL.CommandType, insertSQL.Text, insertSQL.ParameterTypes);
 				try
 				{
-					binder.BindValues(insert);
-					session.Batcher.ExecuteNonQuery(insert);
+					// prepare and execute the insert
+					var insert = session.Batcher.PrepareCommand(insertSql.CommandType, insertSql.Text, insertSql.ParameterTypes);
+					try
+					{
+						binder.BindValues(insert);
+						session.Batcher.ExecuteNonQuery(insert);
+					}
+					finally
+					{
+						session.Batcher.CloseCommand(insert, null);
+					}
 				}
-				finally
+				catch (DbException sqle)
 				{
-					session.Batcher.CloseCommand(insert, null);
+					throw ADOExceptionHelper.Convert(session.Factory.SQLExceptionConverter, sqle,
+					                                 "could not insert: " + persister.GetInfoString(), insertSql.Text);
```

Comment:
Yes, I have forced myself putting back those spaces in place after they have been changed to tabs.